.github/workflows/deploy-book.yml

```diff
@@ -145,20 +145,24 @@ jobs:
         ref: ${{ matrix.branch }}
         submodules: 'true'
 
-    - name: Record if branch is preprocessed, for cache key
+    - name: Record if branch is preprocessed for cache
       run: |
         if [ "$BRANCHES_TO_PREPROCESS" == "*" ] || echo "$BRANCHES_TO_PREPROCESS" | tr ' ' '\n' | grep "^${{ matrix.branch }}$"; then
-          echo true > VARS.txt
+          echo true > EXTRA_CACHE_VARS.txt
         else
-          echo false > VARS.txt
+          echo false > EXTRA_CACHE_VARS.txt
         fi
 
+    - name: Record filenames for cache
+      run: |
+         find book/ -type f >> EXTRA_CACHE_VARS.txt
+
     - name: Cache page build
       id: cache-html
       uses: actions/cache@v4
       with:
         path: "book/_build/html"
-        key: html-build-${{ hashFiles('book/**', 'figures/**', 'requirements.txt', 'VARS.txt') }}
+        key: html-build-${{ hashFiles('book/**', 'figures/**', 'requirements.txt', 'EXTRA_CACHE_VARS.txt') }}
```

Comment:
is the `figures/` referring to a folder in the repository? In that case it could be removed because this folder is really not standard, and if it is in the root folder it should contain files which are used in the readme or on repo-level, not in the book itself.

Reply:
OK! removed